ui/src/app/core/settings.service.ts

```diff
@@ -85,11 +106,22 @@ export class SettingsService {
     return this.localStorage.setItem('settings', JSON.stringify(this.currentSettings));
   }
 
-  private createEmptySettingsForProject(projectId: string): ProjectSettings {
+  createEmptySettingsForProject(projectId: string): ProjectSettings {
+    const capabilities = this.capabilitiesService.getCapabilitiesSynchronous();
+    let filters: Filter[] = [];
+    if (capabilities.queryExtensions) {
+      capabilities.queryExtensions.forEach((f) => {
+        const dataType = queryDataTypes.has(f) ? queryDataTypes.get(f) : queryExtensionsDataTypes.get(f);
```

Comment:
Since we're iterating through queryExtensions don't we know this field is in `queryExtensionsDataTypes` and not `queryDataTypes` (otherwise it wouldn't be an extension)? Might be missing something though.